### PR TITLE
161836698 operator style and misc fixes pt1

### PR DESCRIPTION
**Pull request by philosotractor**
# Added
* cypress: Initial operator e2e test module
* views: own-services set some element-ids to enable test automation
* views: transport-operator set some element-ids to enable test automation
* ui: form fields support for setting DOM element id
  
# Fixed
* views: fix operator deleting, button not displayed or working
   
# Changed
* views, controller: renamed :business-id-exists? and code formatting

### CODE REVIEW

File: cypress/integration/operator.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+        // Uncomment these when YTJ service is stubbed and creation is tested fully
+        // cy.get('#select-operator-at-own-services').click();
+        // cy.contains(testOp1.name).should('not.exist');
+        // cy.contains('Omat palvelutiedot').click();
+
+        cy.get('#btn-add-new-transport-operator').click();
+        cy.get('#btn-submit-business-id').as('btnSubmit')
+        cy.get('@btnSubmit').should('be.disabled')
+        cy.get('#input-business-id').type('1')
```
Comment (rckopteri) [inline cypress/integration/operator.js]: admirable systematic approach, but you could add 1234567- with one line and check disabled status.

Comment (philosotractor) [reply cypress/integration/operator.js]: Yes, but the world would be less perfect. 
Let's do  like that when execution time or case length are too long :)

File: ote/src/cljs/ote/ui/form_fields.cljs [hunk 2]
```diff
@@ -1091,13 +1094,15 @@
...
-     }]]
+    (merge
+      (when element-id {:id element-id})
+      {:label label
+       :primary primary
+       :secondary secondary
+       :on-click #(on-click)
+       :disabled disabled
+       :style style
+       })]]
```
Comment (rckopteri) [inline ote/src/cljs/ote/ui/form_fields.cljs]: Trailing parentheses

File: ote/src/cljs/ote/views/transport_operator_ytj.cljs [hunk 3]
```diff
@@ -143,12 +146,18 @@
...
+      {:element-id "input-operator-name"
+       :name :heading-business-id
+       :label (str (tr [:organization-page :business-id-heading]) " " (::t-operator/business-id operator))
+       :type :text-label
+       :h-style :h2}
+
+      {:name :heading2
+       :label (if ytj-company-names-found?
+                (tr [:organization-page :business-id-and-aux-names])
+                "Toiminimi")
```
Comment (rckopteri) [inline ote/src/cljs/ote/views/transport_operator_ytj.cljs]: Do we have translation for Toiminimi?

File: ote/src/cljs/ote/app/controller/transport_operator.cljs [hunk 4]
```diff
@@ -453,18 +453,21 @@
...
   (process-event [{business-id :business-id} app]
-    (let [business-id-validation-error (validation/validate-rule :business-id nil business-id)]
-      (when (nil? business-id-validation-error)
-        (comm/get! (str "transport-operator/ensure-unique-business-id/" business-id)
-                   {:on-success (send-async! ->EnsureUniqueBusinessIdResponse)}))
+    (let [bid-validation-error (validation/validate-rule :business-id nil business-id)
+          resolve-bid-uniqueness-state (fn [state bid-validation-error]
+                                         (if (nil? bid-validation-error)
+                                           (do
+                                             (comm/get! (str "transport-operator/ensure-unique-business-id/" business-id)
```
Comment (jkorpelin) [inline ote/src/cljs/ote/app/controller/transport_operator.cljs]: Looks scary that there are side effects in the right side of a let binding, might be easy to miss?

File: ote/src/cljs/ote/views/transport_operator_ytj.cljs (hunk 3 above)
Comment (philosotractor) [reply ote/src/cljs/ote/views/transport_operator_ytj.cljs]: Translation in coming pull request.

File: ote/src/cljs/ote/ui/form_fields.cljs (hunk 2 above)
Comment (philosotractor) [reply ote/src/cljs/ote/ui/form_fields.cljs]: Will fix in next PR

File: ote/src/cljs/ote/app/controller/transport_operator.cljs (hunk 4 above)
Comment (philosotractor) [reply ote/src/cljs/ote/app/controller/transport_operator.cljs]: The reason is structuring code in readable blocks.
Conder that a func call requires n args, but those are dynamic and calculated. Code to resolve the args can be embedded to the func call, or alternatively resolved earlier and bound to a value.
It is not possible to describe complex logic in a maintainable way without this kind of structure - although alternative suggestions are always welcome :)

**Pull request merged**